src/View/Helper/FormHelper.php

```diff
@@ -306,6 +324,20 @@ public function control($fieldName, array $options = [])
                         $options['label'] = $this->injectClasses('form-check-label', (array)$options['label']);
                     }
                 }
+
+                if ($options['type'] === 'multicheckbox') {
+                    $options['templates']['label'] = $this->templater()->get('multicheckboxLegend');
```

Comment:
Core uses `label` tag for the common title for multi checkboxes, why do we need to change it to legend?

Reply:
Are you referring to the generated markup, or the template name? The `legend` markup is the correct thing to use because the wrapper is a `fieldset`. And while the bootstrap docs are using `fieldset` for radios only, I think using it for multi checkboxes makes sense too.

Reply:
I am referring to the generated markup. If I comment this whole `if` block the markup generated looks good enough with the `<label>` tag. While using `<legend>` might be semantically correct I wonder if deviating from core is necessary when there are no display issues.

Reply:
Okay, display of "select (checkbox, inline)" does have a slight problem.

Reply:
The same would apply when switching radios to bootstraps suggested markup.

I guess from an accesibility perspective, `label` would work in a `fieldset` too, and it would be possible to for example add a `d-block` class to fix the layout issue. `legend` however is already included in the bootstrap CSS and the docs suggest its use. Then again, in a non-horizontal layout, the `legend` looks off, which however could be fixed by applying the `col-form-label pt-0` classes that are already used for horizontal align.

Reply:
For horizontal form (`/pages/horizontal-align#select-checkbox-anchor`), with your change we end up with `<legend>` tag nested inside inside `<div>` tag. That's invalid markup. 

`validator.w3.org` says `Element legend not allowed as child of element div in this context.`

Reply:
Yeah, the specs are pretty resrictive there (for no good reason IMHO), the problem is that styling won't work properly when using the "valid" markup.

https://github.com/twbs/bootstrap/issues/24938

I'm not married to the idea of using fieldsets, I think it would be nice if we could get it working, but I'd be fine with using divs and labels for now if they have a better pros/cons ratio.

Reply:
Reading that issue I fail to see why they can't achieve same display while still having valid HTML but let's not get into that.

Given that core uses `<label>` tag by default and using `<legend>` generates invalid markup I would rather stick with what core markup and see if we can just fix the display issue by applying required classes to the `<label>` tag.

I tried checking what other frameworks do and found this e.g. of [horizontal form](https://bulma.io/documentation/form/general/#horizontal-form) using bulma which also uses `<label>` for the additional label/title for checkbox/radio groups and doesn't using `<fieldset>`/`<legend>`.

Reply:
> Reading that issue I fail to see why they can't achieve same display while still having valid HTML but let's not get into that.

Styling the `legend` element is really difficult, especially in the context of a `fieldset`, browsers do apply all sorts of annoying restrictions there. The float workaround only works in Firefox, and AFAIK the only somewhat realiable way of positioning would be by using absolute positioning, but that would come with its own issues, especially with regards to responsiveness.

I can't really believe that we still have to deal with this, this behavior is annoying people for decades already, I bet this is in one way or another Internet Explorer's fault, like anything else that sucks on the internet :grimacing: 

Anyways... let's see how using divs and labels works out, I'll give it a try later today.